src/ongui.cpp

```diff
@@ -430,6 +432,19 @@ void MainFrame::togglePTT(void) {
 
     if (g_tx)
     {
+        // Trigger end of TX processing. This causes us to wait for the remaining samples
+        // to flow through the system before toggling PTT
+        int sample = g_outfifo1_empty;
+        endingTx = true;
+        //while (g_outfifo1_empty == sample) wxThread::Sleep(20);
+        int i = 0;
+        while ((i < 20) && (g_outfifo1_empty == sample)) {
+            fprintf(stderr, "sample: %d g_outfifo1_empty: %d\n", sample, g_outfifo1_empty);
+            i++;
+            wxThread::Sleep(50);
+        }
+        fprintf(stderr, "sample: %d g_outfifo1_empty: %d\n", sample, g_outfifo1_empty);
+        
```

Comment:
I figure we just need to:
1. Answer the question on if the fractional FIFO code helps the cut off issue.
1. Clean this up, ie remove the fprintfs and decide on the time out (I suggest 1000ms)
... and we're done :+1: 

Is there anything else you think we need to do?  It's been a real joint effort!

Reply:
Nope, that sounds about it. I removed the debug prints that I could see, and the timeout seems to already be 1000ms based on the current logic (50ms * 20 tries). 👍